src/commands/deploy.ts

```diff
@@ -28,82 +32,104 @@ export default class Deploy extends Command {
-
-      await instantiate({
-        conf,
-        signer,
-        admin,
-        sequence: 1 + sequence,
-        contract: args.contract,
-        codeId,
+      const lcd = new LCDClient(connections(flags.network));
+      const signer = await getSigner({
         network: flags.network,
-        instanceId: flags['instance-id'],
-        refsPath: flags['refs-path'],
+        signerId: flags.signer,
+        keysPath: flags['keys-path'],
         lcd,
       });
-    }
-
-    await this.config.runCommand('contract:generateClient', [
-      args.contract,
-      '--build-schema',
-    ]);
-
-    if (!flags['no-sync']) {
-      await this.config.runCommand('sync-refs', [
-        '--refs-path',
-        flags['refs-path'],
-        '--dest',
-        flags['frontend-refs-path'],
+
+      if (conf.deployTask) {
+        await this.config.runCommand('task:run', [
+          conf.deployTask,
+          '--signer',
+          flags.signer,
+          '--network',
+          flags.network,
+          '--refs-path',
+          flags['refs-path'],
+          '--config-path',
+          flags['config-path'],
+          '--keys-path',
+          flags['keys-path'],
+        ]);
+      } else {
+        // Store sequence to manually increment after code is stored.
+        const sequence = await signer.sequence();
+
+        const codeId = await storeCode({
+          lcd,
+          conf,
+          signer,
+          noRebuild: flags['no-rebuild'],
+          contract: args.contract,
+          network: flags.network,
+          refsPath: flags['refs-path'],
+          useCargoWorkspace: globalConfig.useCargoWorkspace,
+        });
+
+        // pause for account sequence to update.
+        // eslint-disable-next-line no-promise-executor-return
+        await new Promise((r) => setTimeout(r, 1000));
+
+        const admin = flags['admin-address']
+          ? flags['admin-address']
+          : signer.key.accAddress;
+
+        await instantiate({
+          conf,
+          signer,
+          admin,
+          sequence: 1 + sequence,
+          contract: args.contract,
+          codeId,
+          network: flags.network,
+          instanceId: flags['instance-id'],
+          refsPath: flags['refs-path'],
+          lcd,
+        });
+      }
+
+      await this.config.runCommand('contract:generateClient', [
+        args.contract,
+        '--build-schema',
       ]);
-    }
+
+      if (!flags['no-sync']) {
+        await this.config.runCommand('sync-refs', [
+          '--refs-path',
+          flags['refs-path'],
+          '--dest',
+          flags['frontend-refs-path'],
+        ]);
+      }
+    };
+
+    // Error check to be performed upon each backtrack iteration.
+    const errorCheck = () => {
+      if (existsSync('contracts') && !existsSync(join('contracts', args.contract))) {
+        TerrainCLI.error(
+          `Contract '${args.contract}' not available in 'contracts/' directory.`,
+        );
+      }
+    };
```

Comment:
I wonder if this could be pulled out into a separate file since this exact error check is used in a few places.